src/runtime/wasm.ts

```diff
@@ -67,32 +129,58 @@ export abstract class StringEncoder {
     throw new Error(`Unsupported type for StringEncoder: ${type}`);
   }
 
-  abstract encodeSingleton(entity: Storable): string;
+  protected abstract encodeStorable(storable: Storable);
+
+  encodeSingleton(storable: Storable): DynamicBuffer {
+    this.buf = new DynamicBuffer();
+    this.encodeStorable(storable);
+    const buf = this.buf;
```

Comment:
Nit: Give a different name to make more readable compared to this.buf.

Reply:
Done.